### PR TITLE
build(dependencies): update `typescript` to `4.9.4` and its dependencies to their latest version

### PR DESCRIPTION
**Pull request by LuisMartinez15**
EX-7289

Update `typescript` to `4.9.4` and its dependencies to their latest compatible versions.

This task was done with @tiborux . We've not introduced any new operator or done any changes in the code yet. We should probably start using new operators, such as `satisfies`, in order to benefit from them. With this update we are taking advantage of several performance improvements, including some of them targeted directly at monorepo setups.

### CODE REVIEW

**Comment by ajperezbau** [comment]
Should we update `prettier` to its latest version?
Version `2.8.2` brings support for some `Typescript 4.9` features, like the `satisfies` operator or the new auto accessor syntax.

**Comment by ajperezbau** [comment]
Good job! 👌

**Comment by annacv** [comment]
> 

Tried removing `ts-lint` and upgrading to `prettier 2.8.2` and everything working fine

**Comment by LuisMartinez15** [comment]
@ajperezbau @annacv Removed `tslint` and updated `prettier` to its latest version, thanks!

File: packages/eslint-plugin-x/package.json [hunk 1]
```diff
@@ -28,8 +28,8 @@
     "commit": "git-cz"
   },
   "dependencies": {
-    "@typescript-eslint/eslint-plugin": "~5.48.0",
-    "@typescript-eslint/parser": "~5.48.0",
+    "@typescript-eslint/eslint-plugin": "~5.48.2",
```
Comment (tiborux) [inline packages/eslint-plugin-x/package.json:31]:
```suggestion
    "@typescript-eslint/eslint-plugin": "~5.49.0",
```

File: packages/eslint-plugin-x/package.json [hunk 2]
```diff
@@ -28,8 +28,8 @@
...
+    "@typescript-eslint/parser": "~5.48.2",
```
Comment (tiborux) [inline packages/eslint-plugin-x/package.json:32]:
```suggestion
    "@typescript-eslint/parser": "~5.49.0",
```

File: packages/x-components/CHANGELOG.md [hunk 3]
```diff
@@ -1,29 +1,28 @@
 # Change Log
```
Comment (tiborux) [inline packages/x-components/CHANGELOG.md:1]: I thin that updating the branch with main, it will solve the changes in this file

**Review by tiborux — COMMENTED** [review]
update it with main

**Comment by support-empathy** [comment]
This is a link to Empathy Platform Docs Portal preview in a development environment. 
 ⚠️ Only for Empathy members. We’re sorry for the inconvenience!
 [Check out EPDocs preview](https://docs-dev.empathy.co/preview/x-993/index.html)

**Comment by support-empathy** [comment]
This is a link to Empathy Platform Docs Portal preview in a development environment. 
 ⚠️ Only for Empathy members. We’re sorry for the inconvenience!
 [Check out EPDocs preview](https://docs-dev.empathy.co/preview/x-993/index.html)

**Review by ajperezbau — APPROVED** [review]

**Review by ajperezbau — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**